### PR TITLE
OVN provisioning changes

### PR DESCRIPTION
**Pull request by pecameron**
Move cluster provisioning from ovn-setup.sh to ovn-setup.yaml
The remainder of ovn-setup.sh collects info to setup the ovn-config
configmap.

After the cluster is installed, kubectl create -f yaml/ovn-setup.yaml

On the master host run ./ansible/scripts/setup.sh

Then startup the daemonsets.

Signed-off-by: Phil Cameron <pcameron@redhat.com>

### CODE REVIEW

**Comment by pecameron** [comment]
@rajatchopra @dcbw @squeed PTL

File: dist/yaml/ovn-setup.yaml [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+# Create the project, service account and policies.
+# ovnkube interacts with kubernetes and the environment
+# must be properly set up.
+# 
+# This provisioning is done as part of installation after the cluster is
+# up and before the ovn daemonsets are created.
+
+---
+apiVersion: project.openshift.io/v1
+kind: Project
```
Comment (rajatchopra) [inline dist/yaml/ovn-setup.yaml]: For kubernetes, a namespace object is sufficient. We should split this file into two - one for plain kubernetes, and another for openshift-overrides.

File: dist/yaml/ovnkube-master.yaml [hunk 2]
```diff
@@ -47,6 +47,8 @@ spec:
         # This is the community build
         image: "docker.io/ovnkube/ovn-daemonset:latest"
         # This is an official redhat build
```
Comment (rajatchopra) [inline dist/yaml/ovnkube-master.yaml]: We should remove the official red hat build stuff.

File: dist/yaml/ovn-setup.yaml (hunk 1 above)
Comment (pecameron) [reply dist/yaml/ovn-setup.yaml]: @rajatchopra All but the last yaml are openshift.io/v1. Are you thinking of putting the last item into another file?

File: dist/yaml/ovnkube-master.yaml (hunk 2 above)
Comment (pecameron) [reply dist/yaml/ovnkube-master.yaml]: @rajatchopra would it be possible to leave the RH in? This is very convenient for me when testing on my cluster, I just adjust the commented lines.

File: dist/yaml/ovn-setup.yaml [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+metadata:
+  annotations:
+    openshift.io/description: ""
+    openshift.io/display-name: ""
+    openshift.io/node-selector: ""
+    openshift.io/sa.scc.mcs: s0:c8,c2
+    openshift.io/sa.scc.supplemental-groups: 1000060000/10000
```
Comment (dcbw) [inline dist/yaml/ovn-setup.yaml]: @pecameron we don't need any of these annotations; we can remove them.

File: dist/yaml/ovn-setup.yaml [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+  name: ovn-kubernetes
+spec:
+  finalizers:
+  - openshift.io/origin
+  - kubernetes
```
Comment (dcbw) [inline dist/yaml/ovn-setup.yaml]: @pecameron we don't need any of thse finalizers either

File: dist/yaml/ovn-setup.yaml [hunk 5]
```diff
@@ -0,0 +1,189 @@
...
+kind: Project
+metadata:
+  annotations:
+    openshift.io/description: ""
+    openshift.io/display-name: ""
+    openshift.io/node-selector: ""
+    openshift.io/sa.scc.mcs: s0:c8,c2
+    openshift.io/sa.scc.supplemental-groups: 1000060000/10000
+  name: ovn-kubernetes
+spec:
```
Comment (dcbw) [inline dist/yaml/ovn-setup.yaml]: @pecameron we don't need any of the finalizers; you can just remove the entire spec: section.

File: dist/yaml/ovn-setup.yaml [hunk 6]
```diff
@@ -0,0 +1,189 @@
...
+#
+# Create the project, service account and policies.
+# ovnkube interacts with kubernetes and the environment
+# must be properly set up.
+# 
+# This provisioning is done as part of installation after the cluster is
+# up and before the ovn daemonsets are created.
+
+---
+apiVersion: project.openshift.io/v1
```
Comment (dcbw) [inline dist/yaml/ovn-setup.yaml]: apiVersion can just be "v1"

File: dist/yaml/ovn-setup.yaml (hunk 1 above)
Comment (dcbw) [reply dist/yaml/ovn-setup.yaml]: @pecameron instead of Project, just do a normal Kubernetes "Namespace"

File: dist/yaml/ovn-setup.yaml [hunk 7]
```diff
@@ -0,0 +1,189 @@
...
+
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: ovn
+  namespace: ovn-kubernetes
+
+---
+apiVersion: authorization.openshift.io/v1
```
Comment (dcbw) [inline dist/yaml/ovn-setup.yaml]: All of the authorization.openshift.io/v1 should instead be "rbac.authorization.k8s.io/v1"

**Comment by pecameron** [comment]
@rajatchopra @dcbw @squeed PTAL

File: dist/yaml/ovnkube-master.yaml (hunk 2 above)
Comment (rajatchopra) [reply dist/yaml/ovnkube-master.yaml]: No problem with the meta data in comments I think.

**Pull request merged**